### ISSUE TITLE
The status update of the pod on the edge node failed.

### ISSUE THREAD

**Issue by lishaokai1995** [kubeedge/kubeedge#4687]
**What happened and what you expected to happen**:
When I use a script to concurrently connect 50 edge nodes to the cloud, there is a phenomenon of frequent switching between the Ready and NotReady states of the nodes (I suspect that this is due to network blocking causing loss of heartbeat requests). At this time, the DeamonSet will start pods on newly connected edge nodes, and when the node status stabilizes, the PodReady and ContainersReady in the pod's Condition are often set to false and not updated. I have read the source code and have some questions. The PatchPodStatus() method in vendor/k8s.io/kubernetes/pkg/util/pod/pod.go will compare the old and new podStatus, and then only patch the changed parts in podStatus. At the same time, the processPatch() method in edge/pkg/metamanager/process.go will first update the patch request content to the database and then send it to the cloud. The problem now is, when encountering unstable network conditions when connecting to edge nodes as described above, will there be a situation where the patch update to the database is successful but the sendToCloud fails? Also, because kubelet only patches the changed content, will it cause some pod statuses to never be updated?
**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):1.20.10
- KubeEdge version(e.g. `cloudcore --version` and `edgecore --version`):cloudcore:1.12.1   edgecore:1.12.2

**Comment by Shelley-BaoYue** [comment]
`when encountering unstable network conditions when connecting to edge nodes as described above, will there be a situation where the patch update to the database is successful but the sendToCloud fails?` 

-> 

Thank u! This exactly is a problem because query pod request from edged will get data from edge database. Maybe we need update the database after patch pod successfully.

**Comment by lishaokai1995** [comment]
> `when encountering unstable network conditions when connecting to edge nodes as described above, will there be a situation where the patch update to the database is successful but the sendToCloud fails?`
> 
> ->


How to confirm if a pod has been successfully patched?

**Comment by lishaokai1995** [comment]
> `when encountering unstable network conditions when connecting to edge nodes as described above, will there be a situation where the patch update to the database is successful but the sendToCloud fails?`
> 
> ->
> 
> Thank u! This exactly is a problem because query pod request from edged will get data from edge database. Maybe we need update the database after patch pod successfully.

How to confirm if a pod has been successfully patched?

**Comment by Shelley-BaoYue** [comment]
> > `when encountering unstable network conditions when connecting to edge nodes as described above, will there be a situation where the patch update to the database is successful but the sendToCloud fails?`
> > ->
> > Thank u! This exactly is a problem because query pod request from edged will get data from edge database. Maybe we need update the database after patch pod successfully.
> 
> How to confirm if a pod has been successfully patched?

`Patch` func in `kubeedge/edge/pkg/metamanager/client/pod.go` will receive the response of patch pod request.

**Comment by lishaokai1995** [comment]
Maybe my understanding of the source code is not sufficient, and I hope to continue to consult with you. I think that the `patch` func in `kubeedge/edge/pkg/metamanager/client/pod.go` will eventually call the `processPatch` func in `edge/pkg/metamanager/process.go`. The `sendToCloud` func  is asynchronous, so we cannot determine whether the patch request has been successfully sent to the cloud.

**Comment by Shelley-BaoYue** [comment]
`sendSync` in `patch` func of `kubeedge/edge/pkg/metamanager/client/pod.go` is synchronized. The `sendToCloud` func exactly is asynchronous, but it doesn't send a response to `patch`. The `patch`  will wait for the response until cloudcore send a response. The whole process is like this: `Patch` in client/pod.go --> processPatch in metamanegr/process.go --> edgecontroller in cloudcore --> processResponse in metamanegr/process.go --> `Patch` in client/pod.go.

**Comment by Shelley-BaoYue** [comment]
hi, @lishaokai1995 , I reconfirmed that patchPod won't update pod data in db, it just record data with type `podpatch`.  The type `pod` data will be updated when edgecontroller-downstream has watched the pod changed in cluster(patch pod successfully) and send new data to edge.    I also pushed a pr #4696 here. it update the pod db when metamanager receive a patchPod response message and it may help when the network is not stable.
